shopelectro/views/catalog.py

```diff
@@ -229,26 +256,28 @@ def template_context(page, tag_titles, tags):
                 'tags': tags,
             }
 
-        page = context['page']
-        page.get_template_render_context = partial(
-            template_context, page, tag_titles, tags)
+        tags = self.get_tags()
+        if tags:
+            tag_titles = models.serialize_tags_to_title(tags)
+            page = context['page']
+            page.get_template_render_context = partial(
+                template_context, page, tag_titles, tags)
 
-        paginated_page = get_paginated_page_or_404(all_products, products_on_page, page_number)
-        total_products = all_products.count()
+        paginated_page = get_paginated_page_or_404(products, products_on_page, page_number)
+        total_products = products.count()
         products = paginated_page.object_list
         if not products:
             raise http.Http404('Page without products does not exist.')
 
         return {
             **context,
             'products_data': merge_products_context(products),
```

Comment:
It will increase quantity of queries too, because you are already doing it in `CategoryPage` class